drivers/SmartThings/zigbee-thermostat/src/popp/init.lua

```diff
@@ -0,0 +1,257 @@
+    attributeId = VIEWING_DIRECTION_ATTR,
+    dataType = data_types.Enum8
+  },
+  eTRVOrientation = {
+    clusterId = Thermostat.ID,
+    attributeId = ETRV_ORIENTATION_ATTR,
+    dataType = data_types.Boolean
+  },
+  regulationSetPointOffset = {
+    clusterId = Thermostat.ID,
+    attributeId = REGULATION_SETPOINT_OFFSET_ATTR,
+    dataType = data_types.Int8
+  },
+  windowOpenFeature = {
+    clusterId = Thermostat.ID,
+    attributeId = WINDOW_OPEN_FEATURE_ATTR,
+    dataType = data_types.Boolean
+  }
+}
+
+local SUPPORTED_MODES = {
+  ThermostatMode.thermostatMode.heat.NAME,
+  ThermostatMode.thermostatMode.eco.NAME
+}
+
+local is_popp_thermostat = function(opts, driver, device)
+  for _, fingerprint in ipairs(POPP_THERMOSTAT_FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+-- Attribute Refresh Function
+local do_refresh = function(driver, device)
+
+  local attributes = {
+    Thermostat.attributes.OccupiedHeatingSetpoint,
+    Thermostat.attributes.LocalTemperature,
+    ThermostatUIConfig.attributes.KeypadLockout,
+    PowerConfiguration.attributes.BatteryVoltage
+  }
+  for _, attribute in pairs(attributes) do
+    device:send(attribute:read(device))
+  end
+
+  -- refresh custom attributes:
+  -- window open state
+  -- external window open state
+  local custom_thermostat_attributes = {
+    VIEWING_DIRECTION_ATTR,
+    common.EXTERNAL_OPEN_WINDOW_DETECTION_ID
+  }
+  for _, attribute in pairs(custom_thermostat_attributes) do
+    device:send(cluster_base.read_manufacturer_specific_attribute(device, Thermostat.ID, attribute, common.MFG_CODE))
+  end
+
+end
+
+-- Device Added Function
+local device_added = function(driver, device)
+  -- Set supported thermostat modes
+  device:emit_event(ThermostatMode.supportedThermostatModes(SUPPORTED_MODES, { visibility = { displayed = false } }))
+end
+
+-- Configuration Function
+local do_configure = function(driver, device)
+  device:send(device_management.build_bind_request(device, Thermostat.ID, driver.environment_info.hub_zigbee_eui))
+  device:send(device_management.build_bind_request(device, PowerConfiguration.ID, driver.environment_info.hub_zigbee_eui))
+  device:send(Thermostat.attributes.LocalTemperature:configure_reporting(device, 5, 300, 10)) -- report temperature changes over 0.1°C
+  device:send(Thermostat.attributes.OccupiedHeatingSetpoint:configure_reporting(device, 5, 300, 50))
+  device:send(PowerConfiguration.attributes.BatteryVoltage:configure_reporting(device, 30, 21600, 1))
+end
+
+local function device_init(driver, device)
+  battery_defaults.build_linear_voltage_init(2.4, 3.2)(driver, device)
+
+  -- Add the manufacturer-specific attributes to generate their configure reporting and bind requests
+  for _, config in pairs(get_cluster_configurations()) do
+    device:add_configured_attribute(config)
+    device:add_monitored_attribute(config)
```

Comment:
this won't do anything unless you also call `device:configure()` inside your `doConfig` handler